lib/iris/__init__.py

```diff
@@ -210,6 +215,7 @@ def __repr__(self):
                           self.netcdf_no_unlimited, self.clip_latitudes)
 
     deprecated_options = {'cell_datetime_objects': 'warning',
+                          'netcdf_no_unlimited': 'error',
```

Comment:
The "error" behaviour doesn't seem right. 😕
If I've been a Good User and taken account of the deprecation warnings then my current, v1.9 compatible code should look something like:
```
with iris.Future(netcdf_no_unlimited=True):
    # Do clever stuff with netCDF files
```
With the behaviour proposed in this PR, when I upgrade to v2.0 my code will get an error. 👎 The whole point of deprecations is to allow seemless migration.
Instead, I should get a deprecation warning about setting `netcdf_no_unlimited=True`. This lets my code continue to run, and I can remove the now unnecessary `with` block in my own good time.

If I've been a Bad User and ignored the deprecation warnings from v1.9, my code might just be:
```
# Do clever stuff with netCDF files that relies on the old behaviour
```
In which case my code is now broken. That's my own silly fault.
Or perhaps my code looks like:
```
with iris.Future(netcdf_no_unlimited=False):
    # Do clever stuff with netCDF files that relies on the old behaviour
```
In which case I should now get an error because I'm trying to set the context to an unsupported value. Again, this is my own silly fault.

In other words, setting netcdf_no_unlimited to True should provoke a warning, but setting it to False should raise an error.

Reply:
👍 

Indeed it should. I notice that is what is documented in the what's new too...

Reply:
I raised the same point on another PR (https://github.com/SciTools/iris/pull/2858) and was convinced that this is acceptable because 2.0 is a major release. I haven't been following all these deprecation PRs, so I'm not sure if there are others which do the same thing.

Reply:
The same behaviour has been added in https://github.com/SciTools/iris/pull/2848

Reply:
Discussed offline with @pelson: I'll put up a separate PR to change the behaviour of the `Future` object so that the 'error' behaviour is to raise an exception if the flag is set to `False` and raise a warning when it is set to `True`.